src/fsharp/FSharp.Core/control.fs

```diff
@@ -529,6 +529,25 @@ namespace Microsoft.FSharp.Control
         // delayPrim = "bindA (return ()) f"
         let delayA f = callA f ()
 
+        // protect an exception in an cancellation workflow and adds it to the given OperationCanceledException
+        // ie creates a new instance, where the new information is added,
+        // if cexn is a TaskCanceledException the type is preserved as well.
+        let protectExn (cexn:OperationCanceledException) (edi:ExceptionDispatchInfo) =
```

Comment:
A better name might be `augmentOperationCancelledException`.  In this file `protect` normally means `protect a function with a try/with`

Reply:
yes makes sense